README.md

```diff
@@ -365,6 +365,32 @@ Of course, the situation in which a class grows this long is strongly discourage
   }
   ```
 
+- If the only goal is to match on the type of the object, do NOT expand fully all the arguments, as it makes refactoring more difficult and the code more error prone.
```

Comment:
The latter is pretty hard to reason about, are there easier heuristics available?